### PR TITLE
CoW: Add warning for chained assignment with fillna

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by phofl** [comment]
@jorisvandenbossche I'd appreciate if you can take a look locally, I can't reproduce these failures on Mac

**Comment by jbrockmendel** [comment]
i get similar-looking failures on mac

**Comment by phofl** [comment]
Arm or intel? Purged environment and still can't reproduce...

**Comment by jbrockmendel** [comment]
intel

**Comment by phofl** [comment]
Hm shit. Thx for checking

**Comment by phofl** [comment]
This is Python version specific unfortunately... 3.10 has a different ref count compared to 3.11

File: pandas/tests/copy_view/test_interp_fillna.py [hunk 1]
```diff
@@ -344,3 +346,16 @@ def test_fillna_inplace_ea_noop_shares_memory(
         assert not np.shares_memory(get_array(df, "b"), get_array(view, "b"))
     df.iloc[0, 1] = 100
     tm.assert_frame_equal(df_orig, view)
+
+
+def test_fillna_chained_assignment(using_copy_on_write):
+    df = DataFrame({"a": [1, np.nan, 2], "b": 1})
+    df_orig = df.copy()
+    if using_copy_on_write:
+        with tm.assert_produces_warning(ChainedAssignmentError):
```
Comment (jorisvandenbossche) [inline pandas/tests/copy_view/test_interp_fillna.py:355]: There is a helper for this, `with tm.raises_chained_assignment_error():`, which will take care of the fact that it doesn't raise a warning on PyPy

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good!

File: pandas/tests/copy_view/test_interp_fillna.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/copy_view/test_interp_fillna.py]: thx

**Pull request merged**